shell/platform/darwin/ios/framework/Source/accessibility_bridge.h

```diff
@@ -35,12 +41,82 @@ class AccessibilityBridge;
  */
 @property(nonatomic, strong) SemanticsObject* parent;
 
+/**
+ * The accessibility bridge that this semantics object is attached to. This
+ * object may use the bridge to access contextual application information.
+ */
+@property(nonatomic, readonly) shell::AccessibilityBridge* bridge;
+
+/**
+ * The accessibility node used to produce this semantics object.
+ */
+@property(nonatomic, readonly) blink::SemanticsNode node;
+
+/**
+ * Updates this semantics object using data from the `node` argument.
+ */
+- (void)setSemanticsNode:(const blink::SemanticsNode*)node NS_REQUIRES_SUPER;
+
+/**
+ * Whether this semantics object has child semantics objects.
+ */
+@property(nonatomic, readonly) BOOL hasChildren;
+
+/**
+ * Direct children of this semantics object. Each child's `parent` property must
+ * be equal to this object.
+ */
+@property(nonatomic, readonly) std::vector<SemanticsObject*>* children;
+
+- (BOOL)willCauseLayoutChange:(const blink::SemanticsNode*)node;
+
+#pragma mark - Designated initializers
+
 - (instancetype)init __attribute__((unavailable("Use initWithBridge instead")));
 - (instancetype)initWithBridge:(shell::AccessibilityBridge*)bridge
                            uid:(int32_t)uid NS_DESIGNATED_INITIALIZER;
 
 @end
 
+/**
+ * The default implementation of `SemanticsObject` for most accessibility elements
+ * in the iOS accessibility tree.
+ *
+ * Use this implementation for nodes that do not need to be expressed via UIKit-specific
+ * protocols (it only implements NSObject).
+ *
+ * See also:
+ *  * TextInputSemanticsObject, which implements `UITextInput` protocol to expose
+ *    editable text widgets to a11y.
+ */
+@interface FlutterSemanticsObject : SemanticsObject
+@end
+
+/**
+ * An implementation of `UITextInput` used for text fields that are not currently focused.
+ *
+ * This class is used by `TextInputSemanticsObject`.
+ */
+@interface InactiveTextInput : UIView<UITextInput>
+
+@property(nonatomic, copy) NSString* text;
+@property(nonatomic, readonly) NSMutableString* markedText;
+@property(readwrite, copy) UITextRange* selectedTextRange;
+@property(nonatomic, strong) UITextRange* markedTextRange;
+@property(nonatomic, copy) NSDictionary* markedTextStyle;
+@property(nonatomic, assign) id<UITextInputDelegate> inputDelegate;
+
+@end
+
+/**
+ * An implementation of `SemanticsObject` specialized for expressing text fields.
+ *
+ * Uses `InactiveTextInput` and `FlutterTextInputView` to express blurred and focused
```

Comment:
Same. We should be specific about the focus type (input vs. accessibility focus) here to avoid future confusion.